src/distrs/pois.jl

```diff
@@ -11,3 +11,12 @@ import .RFunctions:
     poisinvccdf,
     poisinvlogcdf,
     poisinvlogccdf
+
+# generic versions
+poispdf(λ::Real, x::Real) = exp(poislogpdf(λ, x))
+
+function poislogpdf(λ::T, x::T) where {T <: Real}
+    iszero(λ) ? (iszero(x) ? x : T(-Inf)) : x * log(λ) - λ - lgamma(x + 1)
```

Comment:
StatsFuns has the `xlogy` function exactly for this purpose

Reply:
Right.  In fact, I think I added it long, long ago.